packages/plugin-ext/src/main/browser/custom-editors/custom-editor-widget.ts

```diff
@@ -45,7 +45,7 @@ export class CustomEditorWidget extends WebviewWidget implements SaveableSource,
         );
     }
     get saveable(): Saveable {
-        return this._modelRef.object;
+        return this._modelRef?.object;
```

Comment:
@safisa given that `_modelRef` can be `undefined` (if `setRef` is not called) I think we should properly type it, do you mind making the updates?

```ts
protected _modelRef?: Reference<CustomEditorModel>;
```

Reply:
More importantly it doesn't fix but moves the problem downstream: `get saveable` is typed as returning `Saveable`, not `Saveable | undefined`. With this change `undefined` may be returned here, causing problems later down.

Reply:
@vince-fugnitto, @paul-marechal, I agree this change may cause problems later down...

Actually, the fix I did in my Theia extension (``this._modelRef?.object``) is working for me and after reloading the page my custom editors keep working as expected, I can do changes and save. But I retested it again on the attached vscode cats example, the custom editor is restored after the reload but it is not functional, I can't do edits!!!

I will keep my override fix in my extension, but I will close this PR change.
WDYT?

Reply:
> I will keep my override fix in my extension, but I will close this PR change.

Up to you. I don't think we'll merge this change in its current form as it would be more interesting to understand why `_modelRef` isn't set before being accessed.